### DISCUSSION TITLE
Succinct way to get entity ID/version?

### DISCUSSION THREAD

**Discussion by Net5F** [skypjack/entt#787] [Q&A] [answered]
Heyo, I saw that registry.entity()/version() were removed and replaced with entt_traits::to_entity()/to_version(). I see that they work, but the way I'm calling them seems a bit verbose. Not a big deal since I can alias them myself, but I just wanted to check, is there a more succinct way to call them than this?

```
entt::entt_traits<entt::entity>::to_entity(entity);
```

**Comment by skypjack** [comment, marked as answer]
Yeah, they were removed because you don't have to create a registry nor to include it to be able to use the storage class.
There is actually no succint way to do it. The best we can do is to turn it into the `basic_entt_traits` vs `entt_traits` model already in use in the rest of the codebase.
This way, for the default entity type you would have: `entt::entt_traits::to_entity(entity);`.

**Reply by Net5F** [reply]
Cool, thanks for the info!

**Reply by skypjack** [reply]
You're welcome. If you find one of my suggestions viable, feel free to continue the discussion here or to open an issue so that I won't forget about it. 😅

**Comment by skypjack** [comment]
Or we could add an `entt::to_entity` function similar to `entt::to_integral`... 🤔

**Reply by Net5F** [reply]
Something like that would be really nice. I think `entt::to_entity` is already taken by the "get an entity ID given a registry and component" function though, no?
```
const auto entity = entt::to_entity(registry, position);
```
Maybe a terminology change is warranted to get away from both "the entity ID, with the version" and "the entity ID, without the version" being called "entity"? They're very close to the same thing, so it's hard to say. Maybe `entt::to_base_entity` or something? Another option would be to just refer to it in regards to the version, ala `entt::without_version` or `entt::strip_version`.

**Reply by skypjack** [reply]
Yeah, it's too late to rename `entt::entity` to `entt::identifier`. 😅 
We can also overload `entt::to_entity` and let it be. Though, I can see how name clashing can be annoying in this case.
Otherwise, we could rename the _extended_ `to_entity` into `to_identifier`, since that's what it returns.
Dunno, naming things is hard, you know... 🙂 I'm open to suggestions.

**Reply by Net5F** [reply]
Naming things is definitely hard, especially when they're such close concepts haha.
Having the full ID + version be the "entity" concept, and the individual parts be "identifier" and "version" seems reasonable to me. Would renaming to `entt_traits::to_identifier` along with a `entt::to_identifier` and `entt::to_version` be an option?

**Reply by skypjack** [reply]
We have a winner from Discord. 😅 
What if we renamed `entt::to_entity` to `entt::to_id`, then added `entt::to_entity` and `entt::to_version` to mimic the same names from `entt_traits`?
Just ignore the breaking change for a moment. It's an easy one on a rarely used function and we already have a goo ìd amount of breaking changes planned for the next version, so... 🤷‍♂️

**Reply by Net5F** [reply]
Sounds good to me!
